### ISSUE TITLE
pkgcore breaks with bash-4.3

### ISSUE THREAD

**Issue by radhermit** [pkgcore/pkgcore#8]
It appears that bash-4.3 doesn't reinitialize declarations of empty index arrays properly causing issues when the envs are dumped and reloaded. The most often hit issue is when an ebuild uses something that makes use of the estack functionality from eutils.eclass. This adds the string "declare -a _ESTACK_eshopts='()'" to the env to be dumped and on reload bash skips it, causing issues for code trying to append to the array. A quick example of the breakage can be seen when trying to install app-vim/colorschemes.

Upstream appeared to have a similar bug related to empty vars that was fixed in 4.3_p20 so this is probably a similar situation.

We need to figure out a simple testcase so upstream can reproduce and fix this.

**Comment by Enlik** [comment]
```bash
        # OK
        $ x() { declare -a var=(); declare -p var; }; x
        declare -a var='()'
        # not OK
        $ y() { declare -a var='()'; declare -p var; }; y
        bash: declare: var: not found
```

Reported in the thread "declare in a function makes a variable unable to be found with declare -p in some cases."

**Comment by radhermit** [comment]
Link to mailing list thread for reference: http://lists.gnu.org/archive/html/bug-bash/2015-02/msg00060.html

**Comment by radhermit** [comment]
Should be fixed in bash-4.3_p33-r2 in the tree.
